rust/lance/src/index.rs

```diff
@@ -1294,6 +1299,83 @@ mod tests {
         assert_eq!(stats["num_indices"], 1);
     }
 
+    #[tokio::test]
+    async fn test_full_text_search() {
+        let words = ["spotted", "crib", "irritating", "disturbed"];
+
+        let dir = tempdir().unwrap();
+        let schema = Arc::new(Schema::new(vec![Field::new("text", DataType::Utf8, false)]));
+        let mut builder = StringBuilder::new();
+        for word in words.iter() {
+            builder.append_value(word);
+        }
+        let array = builder.finish();
+        let fields = vec![Arc::new(array) as ArrayRef];
+        let record_batch = RecordBatch::try_new(schema.clone(), fields).unwrap();
+        let iter_schema: SchemaRef = schema.clone();
+        let batches = vec![Ok(record_batch)].into_iter();
+        let batch_iterator = RecordBatchIterator::new(batches, iter_schema);
+
+        let mut dataset = Dataset::write(batch_iterator, dir.path().to_str().unwrap(), None)
+            .await
+            .unwrap();
+
+        let mut tokenizer_config = TokenizerConfig::default();
+        tokenizer_config = tokenizer_config
+            .language("English")
+            .unwrap_or(TokenizerConfig::default());
+        tokenizer_config = tokenizer_config.remove_stop_words(false);
+        tokenizer_config = tokenizer_config.stem(false);
+        let params = InvertedIndexParams {
+            with_position: true,
+            tokenizer_config,
+        };
+        dataset
+            .create_index(&["text"], IndexType::Inverted, None, &params, true)
+            .await
+            .unwrap();
+
+        dataset
+            .optimize_indices(&OptimizeOptions {
+                num_indices_to_merge: 0,
+                index_names: None,
+            })
+            .await
+            .unwrap();
```

Comment:
If there's no new data, does optimize indices do anything? Will it in the future? Should you append some new data before running this?

Reply:
it's possible, like merging existing indices. but i think it should do nothing for scalar index & FTS, because they always have only 1 delta index. added adding new data here and make the tests cover more cases